src/cart/crt_context.c

```diff
@@ -225,8 +226,9 @@ crt_context_provider_create(crt_context_t *crt_ctx, crt_provider_t provider, boo
 	D_RWLOCK_WRLOCK(&crt_gdata.cg_rwlock);
 	cur_ctx_num = crt_provider_get_cur_ctx_num(primary, provider);
 	max_ctx_num = crt_provider_get_max_ctx_num(primary, provider);
+	ctx_idx = crt_provider_get_ctx_idx(primary, provider);
```

Comment:
so if anything fails in this function, you don't call crt_provider_put_ctx_idx() to reset that idx back to unused. that seems problematic as you might run out of indexes in the table without any being actually in use.

Reply:
good point will do a better cleanup of other things as well in next push

Reply:
On this one, the failures will cause context_destroy that will do the put

Reply:
well that's still a problem because crt_context_destroy can itself fail and not put the context that you take here (you are not checking the return value from destroy on the error path here in create).

Reply:
on the error path in the create we call context_destroy with force=true (second argument), which should cause context_destroy to force go-through everything ignoring intermediate rcs.  Having said that, force arg was not being used in 1 of the rcs, so fixing it next upload